samples/nrf9160/location/src/main.c

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <nrf_modem_at.h>
+#include <modem/lte_lc.h>
+#include <modem/location.h>
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <date_time.h>
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+static struct loc_config config;
+static struct loc_method_config methods[2];
+
+K_SEM_DEFINE(location_event, 0, 1);
+
+static void antenna_configure(void)
+{
+	int err;
+
+	if (strlen(CONFIG_LOCATION_SAMPLE_GNSS_AT_MAGPIO) > 0) {
+		err = nrf_modem_at_printf("%s", CONFIG_LOCATION_SAMPLE_GNSS_AT_MAGPIO);
+		if (err) {
+			printk("Failed to set MAGPIO configuration\n");
+		}
+	}
+
+	if (strlen(CONFIG_LOCATION_SAMPLE_GNSS_AT_COEX0) > 0) {
+		err = nrf_modem_at_printf("%s", CONFIG_LOCATION_SAMPLE_GNSS_AT_COEX0);
+		if (err) {
+			printk("Failed to set COEX0 configuration\n");
+		}
+	}
+}
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+K_SEM_DEFINE(time_update_finished, 0, 1);
+
+static void date_time_evt_handler(const struct date_time_evt *evt)
+{
+	/* Don't care about the event, just indicate that we got it. */
+	k_sem_give(&time_update_finished);
+}
+
+static void current_time_get(void)
+{
+	int err;
+
+	/* If we wait for a while after LTE attach, it's more likely that modem has got the current
+	 * time from the LTE network and no NTP is needed.
+	 */
+	k_sleep(K_SECONDS(1));
+
+	err = date_time_update_async(date_time_evt_handler);
+	if (err) {
+		printk("Requesting the current time failed, error: %d\n", err);
+		return;
+	}
+
+	k_sem_take(&time_update_finished, K_MINUTES(1));
+}
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+static const char *method_string_get(enum loc_method method)
```

Comment:
I've been thinking if we should have this function in the location library API. What do you think? Quite often needed by the clients to print the method either in real output or into logs.

Reply:
yes, and maybe for a service as well.

Reply:
Sounds good to me. But the used service is not indicated in the location event, so not sure if that's needed in the library API? It could be somewhere else?

Reply:
`location_method_str()` added to `location.h`.
Service has a static function `location_core_service_str()` in `location_core`, which can be later added to API if needed.